Cargo.toml

```diff
@@ -22,9 +22,18 @@ std = []
 name = "hex"
 harness = false
 
+[dependencies]
+serde = { version = "1.0", optional = true }
+
 [dev-dependencies]
 criterion = "0.3"
 rustc-hex = "2.0"
 faster-hex = "0.4"
 version-sync = "0.8"
 pretty_assertions = "0.6"
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0"
+
+[package.metadata.docs.rs]
+all-features = true
+rustdoc-args = ["--cfg", "docsrs"]
```

Comment:
Could you put the docs.rs stuff on another PR, as it's totally unrelated to this one?
Would kinda make a small PR but i prefer this to having a PR with two unrelated topics/commits actually :smile_cat:

Reply:
I guess I don't really think it is unrelated. This PR introduces `serde` behind a feature flag and the docrs config is to enable that feature to show up on docs.rs.

Also note the addition in lib.rs:

```rust
#[cfg_attr(docsrs, doc(cfg(feature = "serde")))]
pub mod serde;
```

---

![image](https://user-images.githubusercontent.com/221559/74115108-81440280-4b7b-11ea-8ed0-d0044020bdd0.png)

---

![image](https://user-images.githubusercontent.com/221559/74115121-8dc85b00-4b7b-11ea-99ba-71257d49f826.png)

Reply:
Ok i didn't know this attribute existed, looks nice and useful indeed.
i suppose it's effectively not unrelated, my bad ^^